src/parquet/util/buffer.cc

```diff
@@ -34,18 +34,33 @@ std::shared_ptr<Buffer> MutableBuffer::GetImmutableView() {
   return std::make_shared<Buffer>(this->get_shared_ptr(), 0, size());
 }
 
-OwnedMutableBuffer::OwnedMutableBuffer() :
-    ResizableBuffer(nullptr, 0) {}
+OwnedMutableBuffer::OwnedMutableBuffer(MemoryPool* pool) :
+    ResizableBuffer(nullptr, 0), memory_pool_(pool) {}
+
+OwnedMutableBuffer::~OwnedMutableBuffer() {
+  if (data_) {
+    memory_pool_->Free(data_);
+  }
+}
+
+void OwnedMutableBuffer::Reserve(int64_t new_capacity) {
+  if (!data_ || new_capacity > capacity_) {
+    if (data_) {
+      uint8_t* new_data = memory_pool_->Allocate(new_capacity);
+      memcpy(new_data, data_, size_);
+      memory_pool_->Free(data_);
```

Comment:
I feel it would be better to call `Free(mutable_data_)` here, then assign `data_ = mutable_data_` at the end and avoid `const_const` altogether